kong/init.lua

```diff
@@ -428,6 +496,56 @@ function Kong.balancer()
   runloop.balancer.after()
 end
 
+local function plugins_map_wrapper()
+  local version, err = singletons.cache:get("plugins_map:version",
+                                            PLUGINS_MAP_CACHE_OPTS,
+                                            utils.uuid)
+  if err then
+    ngx_log(ngx.CRIT, "could not ensure plugins_map is up to date: ", err)
+
+    return false
+
+  elseif plugins_map_version ~= version then
+    -- try to acquire the mutex (semaphore)
+
+    local ok, err = plugins_map_semaphore:wait(10)
```

Comment:
Should the timeout be configurable?

Reply:
Indeed there's some relevant discussion about the same kind of behavior at https://github.com/Kong/kong/pull/3782#discussion_r219679253; will update this PR to match that behavior once that's settled and approved.